ethcore/light/src/client/header_chain.rs

```diff
@@ -479,6 +542,49 @@ impl HeaderChain {
 		Ok(pending)
 	}
 
+	/// Generates the specifications for hardcoded sync. This is typically only called manually
+	/// from time to time by a Parity developer in order to update the chain specifications.
+	///
+	/// Returns `None` if we are at the genesis block.
+	pub fn read_hardcoded_sync(&self) -> Option<SpecHardcodedSync> {
+		let mut chts = Vec::new();
+
+		for cht_num in 0.. {
+			let cht = match self.cht_root(cht_num) {
+				Some(cht) => cht,
+				_ if cht_num != 0 => {
+					let h_num = 1 + cht_num as u64 * cht::SIZE;
+					let header = if let Some(header) = self.block_header(BlockId::Number(h_num)) {
+						header
+					} else {
+						panic!("Header of block #{} not found in DB", h_num);
+					};
+
+					let decoded = header.decode();
+
+					let entry: Entry = ::rlp::decode(&self.db.get(self.col, era_key(h_num).as_bytes()).unwrap().unwrap());
+					let total_difficulty = entry.candidates.iter()
+						.find(|c| c.hash == decoded.hash())
+						.expect("no candidate matching block found in DB")
```

Comment:
should rather prove based on the fact that an inconsistent but correctly-loaded database shouldn't be possible.